### PR TITLE
Floating add books button and form overlay

### PR DESCRIPTION
**Pull request by chirag-singhal**
Added floating plus button to add books and overlay form for adding details of the book.

### CODE REVIEW

**Comment by taneliang** [comment]
We're also using the Yarn package manager in this project instead of NPM, so the package-lock.json file should be removed :)

**Comment by chirag-singhal** [comment]
@taneliang Removed the package-lock file.

**Comment by chirag-singhal** [comment]
@manyaagarwal Thanks for mentioning the resources for react-navigation. I will use react-native-paper from now. Added react-native-navigation for modal routing.

File: mobile/screens/AddBook.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { StyleSheet, Image, TouchableOpacity, Alert } from "react-native";
+import AddBookForm from "./AddBookForm";
+import { Portal, Modal } from "react-native-paper";
+
+export default function AddBook(props) {
+  const [visible, setVisible] = useState(false);
+
+  const toggleOverlay = () => {
+    setVisible(!visible);
+  };
```
Comment (taneliang) [inline mobile/screens/AddBook.tsx:12]: This needs to be wrapped in a `useCallback` hook to prevent unnecessary rerenders. I'll change it

Comment (chirag-singhal) [reply mobile/screens/AddBook.tsx]: @taneliang I forgot to remove this, this is not being used

**Comment by taneliang** [comment]
Oh no the indentation is messed up again

**Comment by chirag-singhal** [comment]
It got merged before my format code commit, fixing this

**Pull request merged**